### ISSUE TITLE
Changing object property type with Object.assign()

### ISSUE THREAD

**Issue by quarryman** [facebook/flow#7332]
Given I have value of exact type A and want to transform it to exact type B using Object.assign
```
type Coordinates = {|
  lat: number,
  lon: number,
|}

type RawCoordinates = {|
  lat: number,
  lon: string,
|}

const c1: Coordinates = { lat: 1, lon: 2 }
const c2: RawCoordinates = { lat: 1, lon: 'two' }

const merged: RawCoordinates = Object.assign(c1, c2);

const merged2: RawCoordinates = Object.assign({}, c1, c2);
```
How to avoid flow error using Object.assign/spread/ any other option of changing property type?

[Try Flow](https://flow.org/try/#0PTAEAEDMBsHsHcBQiAuBPADgU1AYVrAE4AmAlgHYCGKWAzqALygDeAPoqKNNQFyjkBXALYAjLIQA0HLrHJ9Bo8VNYBfZOmygASpXj4iZKjXpM207innCxk87L60UhCgHNlaxAGNZj0J4CMfPokFNR0jCxcvKD+EjJyoABMoGre5L6eiXw6egQhRuGmUZYxcXAJAOQo8LAVKchpvkLiLljE2brBhmEmoADyIgBWWJ4oAHSUtLSkLuQAFAFxmQCUANwNPiigzYStxFnanXndxhEDw6MTUzPzzCpLsX6Ja0A)

There are similar issues however they address reassigning object property to the value of the same type, while I need reassign to another type

**Comment by jamesisaac** [comment]
I *think* this would be solved by #7298

**Comment by SamChou19815** [comment]
Going forward, we will likely only supporting object spreads well.
